bookkeeper-server/src/test/java/org/apache/bookkeeper/client/BookKeeperTest.java

```diff
@@ -290,4 +290,16 @@ public void readComplete(int rc, LedgerHandle lh, Enumeration<LedgerEntry> seq,
 
         startBKCluster();
     }
+
+    @Test
+    public void testAutoclosableBooKeeper() throws Exception {
+        ClientConfiguration conf = new ClientConfiguration()
+            .setZkServers(zkUtil.getZooKeeperConnectString());
+        try (BookKeeper bkc = new BookKeeper(conf);
+                LedgerHandle lh = bkc.createLedger(digestType, "testPasswd".getBytes());){
+            for (int i = 0; i < 100; i++) {
+                lh.addEntry("foobar".getBytes());
+            }
+        }
```

Comment:
Please add assertion of ledger closure.

Assert.assertTrue("Ledger should be closed!", bkc.isClosed(ledgerId));